services/venice-router/src/main/java/com/linkedin/venice/router/api/path/VenicePath.java

```diff
@@ -54,69 +47,35 @@ public abstract class VenicePath implements ResourcePath<RouterKey> {
   private Set<String> slowStorageNodeSet = new ConcurrentSkipListSet<>();
   private boolean ignoreSlowStorageNodes = false;
 
-  // Whether the request supports streaming or not
-  private VeniceChunkedResponse chunkedResponse = null;
-  private VeniceResponseDecompressor responseDecompressor = null;
-  private Optional<Map<CharSequence, String>> responseHeaders = Optional.empty();
-
-  private long requestId = -1;
-  private int helixGroupId = -1;
-
-  public VenicePath(
-      String storeName,
-      int versionNumber,
-      String resourceName,
-      boolean smartLongTailRetryEnabled,
-      int smartLongTailRetryAbortThresholdMs,
-      RetryManager retryManager) {
-    this(
-        storeName,
-        versionNumber,
-        resourceName,
-        smartLongTailRetryEnabled,
-        smartLongTailRetryAbortThresholdMs,
-        new SystemTime(),
-        retryManager);
-  }
-
   public VenicePath(
-      String storeName,
-      int versionNumber,
-      String resourceName,
-      boolean smartLongTailRetryEnabled,
-      int smartLongTailRetryAbortThresholdMs,
-      Time time,
-      RetryManager retryManager) {
-    this.resourceName = resourceName;
-    this.storeName = storeName;
-    this.versionNumber = versionNumber;
-    this.smartLongTailRetryEnabled = smartLongTailRetryEnabled;
-    this.smartLongTailRetryAbortThresholdMs = smartLongTailRetryAbortThresholdMs;
-    this.time = time;
+      StoreVersionName storeVersionName,
+      RouterRetryConfig retryConfig,
+      RetryManager retryManager,
+      VeniceResponseDecompressor responseDecompressor) {
+    this.storeVersionName = storeVersionName;
+    this.retryConfig = retryConfig;
     this.retryManager = retryManager;
+    this.responseDecompressor = responseDecompressor;
   }
 
   public synchronized long getRequestId() {
-    if (requestId < 0) {
-      requestId = REQUEST_ID_GENERATOR.getAndIncrement();
-    }
-    return requestId;
+    return -1;
   }
 
   public int getHelixGroupId() {
-    return helixGroupId;
+    return -1;
```

Comment:
This is kind of neat.  So we're reducing allocations by removing the member variables on the super class?  At the expense of having the child classes implement these?

I'm wondering about some of these though, the requestId one, do child classes need their own version of the requestId member?  Could we just rely on the super and a default function for getRequestId?  Or is there some cost saving that I'm missing?

Reply:
Or... maybe we don't actually use request id?

Reply:
The request ID is only uses for HAR requests I believe.

Reply:
I realize the comment line is misplaced.  I mean a few lines up on the getRequestId() function

Reply:
Yeah I thought so. Both the Helix group ID and request ID are used for HAR, IIUC.